dist/tools/doccheck/check.sh

```diff
@@ -74,3 +76,47 @@ then
     echo "${UNDEFINED_GROUPS_PRINT}"
     exit 2
 fi
+
+# Check for groups defined multiple times
+MULTIPLE_DEFINED_GROUPS=$( \
+    for group in $(echo ${DEFINED_GROUPS_UNIQUE} | tr ' ' '\n'); \
+    do \
+        # For each group, generate a line with the pattern:
+        # <number of group definitions>:<group name>
+        # Exclude valid groups (e.g. those defined one time)
+        echo $(echo ${DEFINED_GROUPS} | tr ' ' '\n' | grep -cw "^${group}"):${group} | \
+            grep -v "^1:"; \
+    done
+    )
+
+# Remove the group description from each line containing a group definition.
+# Examples of input lines:
+# boards/nucleo-l152re/doc.txt:@defgroup boards_nucleo-l152re STM32 Nucleo-L152RE
+# boards/nucleo-l152re/include/periph_conf.h: * @defgroup boards_nucleo-l152re
+# Expected output lines patterns:
+# <filepath>: * <group name>
+# <filepath>:<group name>
+ALL_RAW_DEFGROUP=$(git grep '@defgroup' -- '*.h' '*.c' '*.txt' | exclude_filter | \
```

Comment:
This is calling `git grep @defgroup` a second time.

Reply:
I reworked the script so that git grep is only called once (for each) to find ingroup and defgroup uses.